src/host/polyfills/simpleView.ts

```diff
@@ -443,3 +443,28 @@ export function applyScreencastCursorPatch(content: string): string | null {
     const replacementText = "this._canvasContainerElement.style.cursor = 'unset';";
     return replaceInSourceCode(content, pattern, replacementText, KeepMatchedText.InFront);
 }
+
+export function applyScreencastHeadlessPatch(content: string): string | null {
+    // This patch will toggle the DevTools screencast on or off based on if the user is using headless or non-headless mode.
+    // This patch also marks a time stamp if the setting is enabled.
+    const pattern = /this\._enabledSetting\s*=\s*Settings\.Settings\.instance\(\)\.createSetting\('screencastEnabled',\s*true\);/g;
+    const replacementText = "const isHeadless = Root.Runtime.vscodeSettings.isHeadless; this._enabledSetting.set(isHeadless); this._startTime = isHeadless ? performance.now() : null;";
```

Comment:
Haven't looked at the rest of the source code here, but have we confirmed that isHeadless will always be in sync with "enabled" from the toggled property below? Seems odd to key them off two different variables

Reply:
Might be easier to understand the context with all the source code, so here are the relevant code snippets with the two patches applied:
```
constructor() {
    // applyScreencastHeadlessPatch patches here
    const isHeadless = Root.Runtime.vscodeSettings.isHeadless;
    this._enabledSetting.set(isHeadless);
    this._startTime = isHeadless ? performance.now() : null;

    this._enabledSetting = Settings.Settings.instance().createSetting('screencastEnabled', true);
    this._toggleButton = new Toolbar.ToolbarToggle(UIString.UIString('Toggle screencast'), 'largeicon-phone');
    this._toggleButton.setToggled(this._enabledSetting.get());
    this._toggleButton.setEnabled(false);
    this._toggleButton.addEventListener(Toolbar.ToolbarButton.Events.Click, this._toggleButtonClicked, this);
    SDKModel.TargetManager.instance().observeModels(ScreenCaptureModel.ScreenCaptureModel, this);
}

_toggleButtonClicked() {
    const enabled = !this._toggleButton.toggled();

    // applyScreenCastTelemetry
    if (enabled) {
      InspectorFrontendHost.InspectorFrontendHostInstance.recordEnumeratedHistogram('DevTools.ScreencastToggle', 1, 2);
        this._startTime = performance.now();
    } else {
        InspectorFrontendHost.InspectorFrontendHostInstance.recordEnumeratedHistogram('DevTools.ScreencastToggle', 0, 2);
        if (this._startTime) {
            const sessionDuration = performance.now() - this._startTime;
            InspectorFrontendHost.InspectorFrontendHostInstance.recordPerformanceHistogram('DevTools.ScreencastDuration', sessionDuration);
        }
    }

    this._enabledSetting.set(enabled);
    this._onScreencastEnabledChanged();
}
```

The `isHeadless` flag is really just to set whether the screencast is enabled or disabled when the ScreencastApp is constructed.  Otherwise, the synchronization is really between the toggle state of the button (defined as `enabled`) and the toggle state of the setting. That is all handled in the DevTools, so I'm not adding anything that requires constant synchronization. Does that make things more clear + answer your question?

Reply:
Absolutely. Thanks!